src/main/java/frc/robot/subsystems/LEDS/NemesisLED.java

```diff
@@ -1,30 +1,113 @@
+import static edu.wpi.first.units.Units.Meters;
+import static edu.wpi.first.units.Units.MetersPerSecond;
+import static edu.wpi.first.units.Units.Seconds;
+
+import edu.wpi.first.wpilibj.AddressableLED;
+import edu.wpi.first.wpilibj.AddressableLEDBuffer;
+import edu.wpi.first.wpilibj.AddressableLEDBufferView;
+import edu.wpi.first.wpilibj.LEDPattern;
+import edu.wpi.first.wpilibj.LEDPattern.GradientType;
+import edu.wpi.first.wpilibj.util.Color;
 import edu.wpi.first.wpilibj2.command.Command;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
-// import frc.robot.Constants;
-import frc.robot.Constants.LEDPattern;
+import java.util.Map;
 
-/** Control REV Robotics Blinkin LED controller */
 public class NemesisLED extends SubsystemBase {
-  private static Spark m_blinkin;
+  private final AddressableLED led;
+  private final AddressableLEDBuffer buffer;
+  private final AddressableLEDBufferView viewStart;
+  private final AddressableLEDBufferView viewEnd;
+
+  private LEDPattern patternStart;
+  private LEDPattern patternEnd;
+
+  public NemesisLED(int port, int length, int halfWay) {
+    // Initialize the LED hardware
+    led = new AddressableLED(port);
+    buffer = new AddressableLEDBuffer(length);
+    viewStart = buffer.createView(0, halfWay);
+    viewEnd = buffer.createView(halfWay + 1, length - 1);
+    led.setLength(length);
+    led.start();
+
+    patternStart = LEDPattern.kOff;
+    patternEnd = LEDPattern.kOff;
+
+    setDefaultCommand(runOnce(() -> LEDPattern.kOff.applyTo(buffer)));
+  }
+
+  @Override
+  public void periodic() {
+    patternStart.applyTo(viewStart);
+    patternEnd.applyTo(viewEnd);
+
+    led.setData(buffer);
+  }
 
-  public NemesisLED(int id) {
-    m_blinkin = new Spark(id);
+  public Command setNemesisFlow(Color color1, Color color2) {
+    var scrollSpeed = MetersPerSecond.of(3);
+    var stripLength = Meters.of(0.05);
+    var scrollingPattern = LEDPattern.gradient(GradientType.kContinuous, color1, color2);
+    return runOnce(
+        () -> {
+          patternStart = scrollingPattern.scrollAtAbsoluteSpeed(scrollSpeed, stripLength);
+          patternEnd = scrollingPattern.scrollAtAbsoluteSpeed(scrollSpeed, stripLength);
+        });
   }
 
-  /**
-   * Set LED pattern
-   *
-   * @param pattern Desired LED light pattern
-   */
-  public Command setPattern(LEDPattern pattern) {
-    return runOnce(() -> m_blinkin.set(pattern.value));
+  public Command setColor(Color color1, Color color2) {
+    return runOnce(
+        () -> {
+          patternStart = LEDPattern.solid(color1);
+          patternEnd = LEDPattern.solid(color2);
+        });
   }
 
-  /** Turn off LEDs */
-  public Command off() {
-    return runOnce(() -> setPattern(LEDPattern.BLACK));
+  public void setColorVoid(Color color1, Color color2) {
+    patternStart = LEDPattern.solid(color1);
+    patternEnd = LEDPattern.solid(color2);
+  }
+
+  public Command setBlink(Color color1, Color color2, double onTime) {
```

Comment:
Looks like you already have a way to blink using a specified color, per above comment.  Great.